### PR TITLE
[Intermediate Representation] Generate AnalyzeIR

### LINKED ISSUE #976: [Intermediate Representation] Generate AnalyzeIR from SQL

### PR DESCRIPTION
**Pull request by weiguoz**
fix #976

### CODE REVIEW

File: pkg/sql/ir_generator_test.go [hunk 1]
```diff
@@ -193,3 +197,59 @@ INTO sqlflow_models.mymodel;`, testDB, modelDir, nil)
...
+	    train.num_boost_round = 30,
+	    eta = 3.1,
+	    num_class = 3
+	COLUMN sepal_length, sepal_width, petal_length, petal_width
+	LABEL class
+	INTO sqlflow_models.my_xgboost_model;
+	`, testDB, modelDir, nil)
+	a.True(goodStream(stream.ReadAll()))
+
+	// Test generate PredicrIR
```
Comment (tonyyang-svail) [inline pkg/sql/ir_generator_test.go]: `PredicrIR ` => `AnalyzeIR`?

File: pkg/sql/codegen/intermediate_representation.go [hunk 2]
```diff
@@ -117,6 +117,8 @@ type AnalyzeIR struct {
 	// "select ... analyze ... with analyze.plot_type = "bar"",
 	// the Attributes will be {"analyze.plot_type": "bar"}
 	Attributes map[string]interface{}
+	// SQLFlow supports TreeExplainer so far.
```
Comment (tonyyang-svail) [inline pkg/sql/codegen/intermediate_representation.go]: This comment is likely to be out of date. Please change to "Explainer types. For example TreeExplainer.".

The list of supported explainers should be documented elsewhere. :)

File: pkg/sql/codegen_alps.go [hunk 3]
```diff
@@ -188,7 +188,9 @@ func newALPSTrainFiller(pr *extendedSelect, db *DB, session *pb.Session, ds *tra
 
 	// TODO(joyyoj) read feature mapping table's name from table attributes.
 	// TODO(joyyoj) pr may contains partition.
-	fmap := columns.FeatureMap{pr.tables[0] + "_feature_map", ""}
+	fmap := columns.FeatureMap{
```
Comment (tonyyang-svail) [inline pkg/sql/codegen_alps.go:191]: Not sure why we need to change these files. Have you used a code formatter?

File: pkg/sql/codegen_alps.go [hunk 4]
```diff
@@ -670,22 +672,22 @@ func (meta *metadata) getDenseColumnInfo(keys []string, refColumns map[string]*c
 			shape[0] = len(fields)
 			if userSpec, ok := refColumns[ct.Name()]; ok {
 				output[ct.Name()] = &columns.ColumnSpec{
-					ct.Name(),
-					false,
-					shape,
-					userSpec.DType,
-					userSpec.Delimiter,
-					nil,
-					*meta.featureMap}
```
Comment (tonyyang-svail) [inline pkg/sql/codegen_alps.go:679]: Is there a linter complaining we need to explicitly name the field name? If so, please file the change in a separate PR.

Comment (weiguoz) [reply pkg/sql/codegen_alps.go:679]: The go vet complains: `[govet] xxx(struct) composite literal uses unkeyed fields [E]` while I'm editing in another file.
I found that there was not much code to be changed, so I changed them conveniently.

**Review by typhoonzero — APPROVED** [review]
LGTM++

**Pull request merged**